tools/pkg/engine_build_configs/lib/src/build_config.dart

```diff
@@ -0,0 +1,820 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+// This library parses Engine build config data out of the "Engine v2" build
+// config JSON files with the format described at:
+//   https://github.com/flutter/engine/blob/main/ci/builders/README.md
+
+/// Base class for all nodes in the build config.
+base class BuildConfigBase {
+  BuildConfigBase(this.errors);
+
+  /// Accumulated errors. Non-null and non-empty when a node is invalid.
+  final List<String>? errors;
+
+  /// Whether there were errors when loading the data for this node.
+  late final bool valid = errors == null;
+
+  /// Returns the list of errors when the object is not valid. Subclasses
```

Comment:
nit: Perhaps it would read better as:

"Returns an empty list when the object is valid, and errors when it is not.

Subclasses..."

Reply:
Good call. Changed.